### PR TITLE
Improve UX with game not registered or with 0 leaderboard

### PR DESCRIPTION
**Pull request by AlexandreSi**

### CODE REVIEW

**Review by ClementPasteau — APPROVED** [review]
lgtm, maybe a tiny check on wording before merging @4ian

File: newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js [hunk 1]
```diff
@@ -46,103 +49,135 @@ const useFetchLeaderboards = () => {
...
+                                : ''
+                            }`}
+                          />
+                        ))
+                      : [
+                          <SelectOption
+                            disabled
+                            key="empty"
+                            value="empty"
+                            primaryText={''}
```
Comment (4ian) [inline newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js:128]: Should we put a primaryText like "No leaderboards created - create one in the leaderboard manager"?

Comment (AlexandreSi) [reply newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js:128]: It felt weird to me to have 2 disabled options in the select, I thought it was clear enough when there is no option + the helper text below.
But maybe that's just me.

image

File: newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js [hunk 2]
```diff
@@ -46,103 +49,135 @@ const useFetchLeaderboards = () => {
...
+export default React.forwardRef<ParameterFieldProps, ParameterFieldInterface>(
+  function LeaderboardIdField(props, ref) {
+    const leaderboards = useFetchLeaderboards();
+    const [isAdminOpen, setIsAdminOpen] = React.useState(false);
+    const genericExpressionFieldRef = React.useRef<?GenericExpressionField>(
+      null
+    );
+    React.useImperativeHandle(ref, () => ({
+      focus: () => {
+        if (genericExpressionFieldRef.current) {
```
Comment (4ian) [inline newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js:61]: Can we have a ref on SelectField?

File: newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js (hunk 1 above)
Comment (4ian) [reply newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js:128]: Right - can we replace "Choose a leaderboard" by  "No leaderboards [..]"?

File: newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js [hunk 3]
```diff
@@ -46,103 +49,135 @@ const useFetchLeaderboards = () => {
...
+                  >
+                    {leaderboards && !!leaderboards.length
+                      ? leaderboards.map(leaderboard => (
+                          <SelectOption
+                            key={leaderboard.id}
+                            value={`"${leaderboard.id}"`}
+                            primaryText={`${leaderboard.name} ${
+                              leaderboard.id
+                                ? `(${breakUuid(
+                                    leaderboard.id.substring(0, 8)
```
[the remark this answers is not in the document]
Comment (AlexandreSi) [reply newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js:117]: Oh yes, didn't spot that!

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js (hunk 2 above)
Comment (AlexandreSi) [reply newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js:61]: I didn't think it was useful but yes it is when using keyboard, will change it

File: newIDE/app/src/GameDashboard/LeaderboardAdmin/index.js [hunk 4]
```diff
@@ -384,6 +385,17 @@ export const LeaderboardAdmin = ({ onLoading }: Props) => {
     );
   }
   if (apiError && apiError.action === 'leaderboardsFetching') {
+    if (!!project) {
+      return (
+        <CenteredError>
+          <GameRegistration
```
[the remark this answers is not in the document]
Comment (AlexandreSi) [reply newIDE/app/src/GameDashboard/LeaderboardAdmin/index.js:391]: Will do

File: newIDE/app/src/GameDashboard/LeaderboardAdmin/index.js [hunk 5]
```diff
@@ -376,6 +385,20 @@ export const LeaderboardAdmin = ({ onLoading }: Props) => {
       </CenteredError>
     );
   }
+  if (!!displayGameRegistration) {
```
Comment (4ian) [inline newIDE/app/src/GameDashboard/LeaderboardAdmin/index.js:388]: Much clearer :) (and correct)

File: newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js:129]: Oh yes, good idea!

**Pull request merged**